src/components/Tag/Tag.scss

```diff
@@ -52,6 +48,20 @@ $icon-size: rem(16px);
     }
   }
 
+  &.removable {
+    padding-right: 0;
+  }
+
+  &.segmented {
+    > * {
+      max-width: 100%;
```

Comment:
This is so ellipsis will work in the segmented case, otherwise I think the inline element has no bound size and overflow won't kick in